packages/documentsearch/src/tokens.ts

```diff
@@ -13,26 +14,300 @@ export const ISearchProviderRegistry = new Token<ISearchProviderRegistry>(
   '@jupyterlab/documentsearch:ISearchProviderRegistry'
 );
 
+/**
+ * Filter interface
+ */
+export interface IFilter {
+  /**
+   * Filter title
+   */
+  title: string;
+  /**
+   * Filter description
+   */
+  description: string;
+  /**
+   * Default value
+   */
+  default: boolean;
+  /**
+   * Does the filter support replace?
+   */
+  supportReplace: boolean;
+}
+
+/**
+ * Type of filters
+ *
+ * TODO support additional filter type
+ */
+export interface IFiltersType {
+  /**
+   * Filter name: value
+   */
+  [key: string]: boolean;
+}
+
+/**
+ * React search component state
+ */
+export interface IDisplayState {
+  /**
+   * The index of the currently selected match
+   */
+  currentIndex: number | null;
+
+  /**
+   * The total number of matches found in the document
+   */
+  totalMatches: number;
+
+  /**
+   * Should the search be case sensitive?
+   */
+  caseSensitive: boolean;
+
+  /**
+   * Should the search string be treated as a RegExp?
+   */
+  useRegex: boolean;
+
+  /**
+   * The text in the search entry
+   */
+  searchText: string;
+
+  /**
+   * The query constructed from the text and the case/regex flags
+   */
+  query: RegExp | null;
+
+  /**
+   * An error message (used for bad regex syntax)
+   */
+  errorMessage: string;
+
+  /**
+   * Should the focus forced into the input on the next render?
+   */
+  forceFocus: boolean;
+
+  /**
+   * Whether or not the search input is currently focused
+   */
+  searchInputFocused: boolean;
+
+  /**
+   * Whether or not the replace input is currently focused
+   */
+  replaceInputFocused: boolean;
+
+  /**
+   * The text in the replace entry
+   */
+  replaceText: string;
+
+  /**
+   * Whether or not the replace entry row is visible
+   */
+  replaceEntryShown: boolean;
```

Comment:
I would suggest `replaceEntryVisible` to be more in line with some other APIs that have `isVisible` attributes.